org.eclipse.xtext.xbase.lib.tests/src/org/eclipse/xtext/xbase/tests/lib/IterableExtensionsTest.java

```diff
@@ -87,7 +87,7 @@ protected Integer findLast(Iterable<Integer> input, Function1<Integer, Boolean>
 
 	@Override
 	protected Integer last(Iterable<Integer> input) {
```

Comment:
Can you please rename this method `last` also to `lastOrNull`?

Reply:
@szarnekow this method is defined as abstract in the superclass `org.eclipse.xtext.xbase.tests.lib.BaseIterablesIteratorsTest.last(IterableOrIterator)` so I should rename it also in `IteratorExtensionsTest`.
I changed to `lastOrNull` only in `IterableExtensions`, NOT in `IteratorExtensions`.

Reply:
Yes, I saw that. I wonder if we should both keep in symmetry, though. Thoughts?

Reply:
I also thought about that, but how do we "justify" the deprecation for `IteratorExtensions`? Of course, we could use the "symmetry" justification.

Also `head` could be changed accordingly.

Unfortunately, we are forced to change `IterableExtensions.last` only for the clash with Java 21 new API.

Reply:
So far I liked that the extension API for iterables and iterators was equivalent (as far as it goes).

Reply:
Ok, then I'll add lastOrNull to IteratorExtensions and update this PR